protocol/test/amo/oud/oud-token.ts

```diff
@@ -0,0 +1,210 @@
+            .to.revertedWith("ERC20: transfer amount exceeds balance");
+    });
+
+    it("owner can recover tokens", async () => {
+        const amount = 50;
+        await token.addMinter(owner.getAddress());
+        await token.mint(token.address, amount);
+        // await recoverToken(token, amount, token, owner);
+    });
+
+    it("Only specified roles can mint", async () => {
+        const alanAddress: string = await alan.getAddress();
+        const minterAddress: string = await minter.getAddress();
+
+        // mint should fail when no minter set.
+        await expect(token.mint(alanAddress, 10))
+            .to.revertedWithCustomError(token, "CannotMintOrBurn")
+            .withArgs(await owner.getAddress());
+
+        // Only admin can add a minter
+        // await shouldRevertNotOwner(token.connect(alan).addMinter(alanAddress));
+        await expect(token.connect(alan).addMinter(alanAddress)).to.revertedWith("Ownable: caller is not the owner");
+        await expect(token.addMinter(minterAddress))
+            .to.emit(token, "AddedMinter")
+            .withArgs(minterAddress);
+        expect(await token.isMinter(minterAddress)).to.be.true;
+
+        // Only minter can, well mint
+        await token.connect(minter).mint(alanAddress, 10);
+        expect(await token.balanceOf(alanAddress)).equals(10);
+        await expect(token.mint(alanAddress, 10))
+            .to.revertedWithCustomError(token, "CannotMintOrBurn")
+            .withArgs(await owner.getAddress());
+
+        // Only admin can remove a minter
+        // await shouldRevertNotOwner(token.connect(alan).removeMinter(minterAddress));
+        await expect(token.connect(alan).removeMinter(minterAddress)).to.revertedWith("Ownable: caller is not the owner");
+        await expect(token.removeMinter(minterAddress))
+            .to.emit(token, "RemovedMinter")
+            .withArgs(minterAddress);
+
+        expect(await token.totalSupply()).eq(10);
+    });
+
+    it("only specified roles can burn", async () => {
+        const alanAddress: string = await alan.getAddress();
+        const minterAddress: string = await minter.getAddress();
+
+        // mint should fail when no minter set.
+        await expect(token.burn(alanAddress, 10))
+            .to.revertedWithCustomError(token, "CannotMintOrBurn")
+            .withArgs(await owner.getAddress());
+
+        await token.addMinter(minterAddress);
+
+        // Only minter can burn
+        await token.connect(minter).mint(alanAddress, 100);
+        await token.connect(minter).burn(alanAddress, 10);
+        expect(await token.balanceOf(alanAddress)).equals(90);
+        await expect(token.burn(alanAddress, 10))
+            .to.revertedWithCustomError(token, "CannotMintOrBurn")
+            .withArgs(await owner.getAddress());
+
+        expect(await token.totalSupply()).eq(90);
+    });
+
+    it("permit works as expected", async () => {
+
+        const signedPermit = async (
```

Comment:
Hmm this isn't actually doing anything...

did you mean to define these functions a layer up, and then actually *call* 
```
        await testErc20Permit(token, alan, spender, 123);
```
?

Reply:
fixed in [9eca314](https://github.com/TempleDAO/temple/pull/760/commits/9eca3142f7140b0f9ea6a54e2ba5c00a7f272245)